pkg/operation/botanist/component/vpa/crd.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vpa
+
+import (
+	"context"
+	_ "embed"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+)
+
+var (
+	//go:embed templates/crd-verticalpodautoscalers.tpl.yaml
+	verticalPodAutoscalerCRD string
+	//go:embed templates/crd-verticalpodautoscalercheckpoints.tpl.yaml
+	verticalPodAutoscalerCheckpointCRD string
+
+	crdResources map[string]string
+)
+
+func init() {
+	crdResources = map[string]string{
+		"crd-verticalpodautoscalers.yaml":           verticalPodAutoscalerCRD,
+		"crd-verticalpodautoscalercheckpoints.yaml": verticalPodAutoscalerCheckpointCRD,
+	}
+}
+
+type vpaCRD struct {
+	applier  kubernetes.Applier
+	registry *managedresources.Registry
```

Comment:
Why was this `registry` added, what's its purpose? I see that in the only place `NewCRD` is invoked, `registry` is `nil`? Wouldn't it be better to add it when it's actually needed for anything?

Reply:
I expected this question :) It is not yet used as part of this PR, but I will use it later when I open the PR refactoring the VPA Helm chart. The `registry` will be used for managing the VPA CRD when deployed into the shoot. I hope it's fine to keep this commit as part of this PR (the other PR will already be quite large).